src/vaadin-message-list.js

```diff
@@ -59,7 +60,8 @@ class MessageListElement extends ElementMixin(ThemableMixin(PolymerElement)) {
         type: Array,
         value: function () {
           return [];
-        }
+        },
+        observer: '_itemsChanged'
```

Comment:
A single property observer only handles the cases when the whole array is reset: `messageList.items = []`
But it does not cover updating the items with Polymer magic methods, e.g. ` messageList.push('items', {})`.

Please consider using a multi-property observer instead, see [vaadin-menu-bar](https://github.com/vaadin/vaadin-menu-bar/blob/fd7ebd5d34ca18684b32e72143f844c11fdb8071/src/vaadin-menu-bar-buttons-mixin.js#L29-L31) for an example.

One downside of multi-propery observers is that you will not get an old value though.

Reply:
Thanks @web-padawan for the suggestions: we discussed this and decided to observe only the replacement of the whole array (also considering the future transition to Lit).